pkg/util/csi/volume_snapshot.go

```diff
@@ -242,3 +296,597 @@ func patchVSC(ctx context.Context, snapshotClient snapshotter.SnapshotV1Interfac
 
 	return patched, nil
 }
+
+func GetPVForPVC(
+	pvc *corev1api.PersistentVolumeClaim,
+	crClient crclient.Client,
+) (*corev1api.PersistentVolume, error) {
+	if pvc.Spec.VolumeName == "" {
+		return nil, errors.Errorf("PVC %s/%s has no volume backing this claim", pvc.Namespace, pvc.Name)
+	}
+	if pvc.Status.Phase != corev1api.ClaimBound {
+		// TODO: confirm if this PVC should be snapshotted if it has no PV bound
+		return nil, errors.Errorf("PVC %s/%s is in phase %v and is not bound to a volume", pvc.Namespace, pvc.Name, pvc.Status.Phase)
+	}
+	pv := &corev1api.PersistentVolume{}
+	err := crClient.Get(context.TODO(), crclient.ObjectKey{Name: pvc.Spec.VolumeName}, pv)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get PV %s for PVC %s/%s", pvc.Spec.VolumeName, pvc.Namespace, pvc.Name)
+	}
+	return pv, nil
+}
+
+func GetPodsUsingPVC(
+	pvcNamespace, pvcName string,
+	crClient crclient.Client,
+) ([]corev1api.Pod, error) {
+	podsUsingPVC := []corev1api.Pod{}
+	podList := new(corev1api.PodList)
+	if err := crClient.List(
+		context.TODO(),
+		podList,
+		&crclient.ListOptions{Namespace: pvcNamespace},
+	); err != nil {
+		return nil, err
+	}
+
+	for _, p := range podList.Items {
+		for _, v := range p.Spec.Volumes {
+			if v.PersistentVolumeClaim != nil && v.PersistentVolumeClaim.ClaimName == pvcName {
+				podsUsingPVC = append(podsUsingPVC, p)
+			}
+		}
+	}
+
+	return podsUsingPVC, nil
+}
+
+func GetPodVolumeNameForPVC(pod corev1api.Pod, pvcName string) (string, error) {
+	for _, v := range pod.Spec.Volumes {
+		if v.PersistentVolumeClaim != nil && v.PersistentVolumeClaim.ClaimName == pvcName {
+			return v.Name, nil
+		}
+	}
+	return "", errors.Errorf("Pod %s/%s does not use PVC %s/%s", pod.Namespace, pod.Name, pod.Namespace, pvcName)
+}
+
+func Contains(slice []string, key string) bool {
+	for _, i := range slice {
+		if i == key {
+			return true
+		}
+	}
+	return false
+}
+
+func IsPVCDefaultToFSBackup(pvcNamespace, pvcName string, crClient crclient.Client, defaultVolumesToFsBackup bool) (bool, error) {
```

Comment:
This one looks like a common function, not only for CSI, can we move it to `podvolume` package?

Reply:
Sure.